promql/promqltest/test.go

```diff
@@ -693,29 +816,79 @@ func (ev *evalCmd) expectMetric(pos int, m labels.Labels, vals ...parser.Sequenc
 	ev.expected[h] = entry{pos: pos, vals: vals}
 }
 
+// validateExpectedAnnotations validates expected messages and regex match actual annotations.
+func validateExpectedAnnotations(expr string, expectedAnnotations []expectCmd, actualAnnotations []string, line int, annotationType string, annos annotations.Annotations) error {
```

Comment:
[nit] Might be good to make it clear that some of these parameters only contain one kind of annotation, and that the last one contains all annotations:

```suggestion
func validateExpectedAnnotationsOfType(expr string, expectedAnnotationsOfType []expectCmd, actualAnnotationsOfType []string, line int, annotationType string, allAnnos annotations.Annotations) error {
```

Reply:
Done!